src/App.tsx

```diff
@@ -3,29 +3,96 @@ import './App.scss';
 import './styles/general.scss';
 import { TodoList } from './components/TodoList';
 import { CurrentUser } from './components/CurrentUser';
+import { getTodos } from './api/api';
 
 interface State {
   selectedUserId: number;
+  visibleTodos: Todo[];
+  query: string;
+  sortBy: string;
 }
 
 class App extends React.Component<{}, State> {
   state: State = {
     selectedUserId: 0,
+    visibleTodos: [],
+    query: '',
+    sortBy: 'all',
+  };
+
+  async componentDidMount() {
+    const todos = await getTodos();
+
+    this.setState({ visibleTodos: todos });
+  }
+
+  handleChangeUserId = (id: number) => {
+    this.setState({
+      selectedUserId: id,
+    });
+  };
+
+  handlerClear = () => {
+    this.setState({
+      selectedUserId: 0,
+    });
+  };
+
+  getVisibleTodos = () => {
+    const { query, visibleTodos, sortBy } = this.state;
+
+    let todos;
+
+    switch (sortBy) {
+      case 'active':
+        todos = visibleTodos.filter(todo => !todo.completed);
+        break;
+      case 'completed':
+        todos = visibleTodos.filter(todo => todo.completed);
+        break;
+      default:
+        todos = visibleTodos;
+    }
+
+    return todos.filter(todo => {
+      const { title } = todo;
+
+      return title.toLowerCase().includes(query.toLocaleLowerCase());
```

Comment:
create a separate variable to `query.toLowerCase()`, no need to do it on every iteration